webui/react/src/hooks/useInputEscape.ts

```diff
@@ -0,0 +1,270 @@
+  focused: boolean,
+  inputRef: React.RefObject<HTMLInputElement | AntdInputRef | RefSelectProps>,
+  event: KeyboardEvent,
+  handleFocused: (focused: boolean) => void,
+) => {
+  if (focused && event.key === 'Escape') {
+    event.stopPropagation();
+    inputRef.current?.blur();
+    getOverlayAndMenuBodyElement()?.menuBody?.focus();
+    handleFocused(false);
+  }
+};
+
+const onClick = (
+  blurred: boolean,
+  focused: boolean,
+  inputRef: React.RefObject<HTMLInputElement | AntdInputRef>,
+  event: MouseEvent,
+  handleFocused: (focused: boolean) => void,
+) => {
+  const overlayClicked =
+    (event.target as HTMLElement).className === getOverlayAndMenuBodyElement()?.overlay?.className;
+
+  if (blurred && focused && overlayClicked) {
+    event.stopPropagation();
+    handleFocused(false);
+  } else if (focused && overlayClicked) {
+    event.stopPropagation();
+    inputRef.current?.blur();
+    handleFocused(false);
+  }
+};
+
+const onClickSelect = (
+  event: MouseEvent,
+  isOpen: boolean,
+  hasOpened: boolean,
+  setHasOpened: (hasOpened: boolean) => void,
+) => {
+  const overlayClassname = getOverlayAndMenuBodyElement()?.overlay?.className;
+  if (isOpen && (event.target as HTMLElement).className === overlayClassname) {
+    event.stopPropagation();
+  }
+
+  if (hasOpened && !isOpen && (event.target as HTMLElement).className === overlayClassname) {
+    event.stopPropagation();
+    // If hasOpened is true in this instance
+    // then the event above will close the
+    // currently open Modal or Menu so we must stop the
+    // event.
+    setHasOpened(false);
+  }
+};
+
+export const useInputNumberEscape = (
+  ref: React.ForwardedRef<HTMLInputElement>,
+  onBlur?: <HTMLInputElement>(
+    e: React.FocusEvent<HTMLInputElement> | React.KeyboardEvent<HTMLInputElement>,
+    previousValue?: string,
+    tagID?: string,
+  ) => void,
+): InputNumberEscape => {
+  const inputRef = React.createRef<HTMLInputElement>();
+  useImperativeHandle(ref, () => inputRef?.current as HTMLInputElement);
+
+  const input = inputRef?.current;
+
+  const [focused, setFocused] = useState(false);
+  const [blurred, setBlurred] = useState(false);
+
+  useEffect(() => {
+    const handleEsc = (event: KeyboardEvent) => {
+      onEsc(focused, inputRef, event, setFocused);
+    };
+
+    const handleClick = (event: MouseEvent) => {
+      onClick(blurred, focused, inputRef, event, setFocused);
+    };
+
+    window.addEventListener('keydown', handleEsc, true);
+    window.addEventListener('click', handleClick, true);
+    return () => {
+      window.removeEventListener('keydown', handleEsc, true);
+      window.removeEventListener('click', handleClick, true);
+    };
+  }, [blurred, focused, input, inputRef]);
+
+  const handleBlur = <HTMLInputElement>(
+    e: React.FocusEvent<HTMLInputElement> | React.KeyboardEvent<HTMLInputElement>,
+    previousValue?: string,
+  ) => {
+    setBlurred(true);
+    if (onBlur) onBlur(e, previousValue);
```

Comment:
sugg: `onBlur?.(e, previousValue);`
You can make this same change elsewhere too.